dirs/dirs.go

```diff
@@ -205,6 +280,19 @@ func StripRootDir(dir string) string {
 	return "/" + result
 }
 
+// DataHomeGlobs returns a slice of globbing expressions for the snap directories in use.
+func DataHomeGlobs(opts *SnapDirOptions) []string {
+	snapHomeDirsMu.Lock()
+	defer snapHomeDirsMu.Unlock()
+	if opts == nil {
```

Comment:
This allocation looks pointless since the zero value of the structure means that we could just return `snapDataHomeGlob` directly.

How about:

``` go
if opts != nil && opts.HiddenSnapDataDir {
   return hiddenSnapDataHomeGlob
}

return snapDataHomeGlob
```